domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmAddHandler.java

```diff
@@ -652,11 +652,14 @@ private void addSSLServices(OperationContext context, ModelNode ssl, ModelNode t
 
     private void addKeyManagerService(OperationContext context, ModelNode ssl, ServiceName serviceName,
                                       ServiceTarget serviceTarget) throws OperationFailedException {
-
-        char[] keystorePassword = KeystoreAttributes.KEYSTORE_PASSWORD.resolveModelAttribute(context, ssl).asString().toCharArray();
         final ServiceBuilder<AbstractKeyManagerService> serviceBuilder;
 
-        String provider = KeystoreAttributes.KEYSTORE_PROVIDER.resolveModelAttribute(context, ssl).asString();
+        ModelNode keystorePasswordNode = KeystoreAttributes.KEYSTORE_PASSWORD.resolveModelAttribute(context, ssl);
+        char[] keystorePassword = keystorePasswordNode.isDefined() ? keystorePasswordNode.asString().toCharArray() : null;
+
+        ModelNode providerNode = KeystoreAttributes.KEYSTORE_PROVIDER.resolveModelAttribute(context, ssl);
+        String provider = providerNode.isDefined() ? providerNode.asString() : null;
```

Comment:
This isn't a request for a change, just an FYI. These can be:

```
String keystorePasswordString = KeystoreAttributes.KEYSTORE_PASSWORD.resolveModelAttribute(context, ssl).asStringOrNull();
char[] keystorePassword = keystorePasswordString != null ? keystorePasswordString.toCharArray() : null;

String provider = KeystoreAttributes.KEYSTORE_PROVIDER.resolveModelAttribute(context, ssl).asStringOrNull();
```

The keystorePassword one it's no big difference but the provider one is quite a bit simpler.

You can use asXXXOrNull() variants for Boolean, Integer, Long etc.

These are new in DMR, added a couple months ago.